Add acceptance test that tests Saga and Outbox together

DavidBoike opened the pull request:
While working on a Raven release, I discovered we don't have an acceptance test that tests a Saga and the Outbox together, nor do we really have one that puts a saga through multiple steps with a possibility of concurrency issue. So I was passing all tests but still having a concurrency issue where successive updates to the saga data were overriding each other. This test covers that case.

andreasohlund commented:
Would it be confusing to have a test with ”outbox” in the name fail even when there is no outbox enabled? 

What would be a better name? When multiple messages update a saga => should not overwrite?

DavidBoike commented:
I was thinking about that after I was forced to do something about enabling the outbox. What if I renamed it to something like When_handling_concurrent_messages and split it into two test cases, one of which did not use the Outbox, and the other one does but makes the call to ignore it if not available?

andreasohlund commented:
I like it

On Tue, 11 Jun 2019 at 20:59, David Boike <notifications@github.com> wrote:

> I was thinking about that after I was forced to do something about
> enabling the outbox. What if I renamed it to something like
> When_handling_concurrent_messages and split it into two test cases, one of
> which did not use the Outbox, and the other one does but makes the call to
> ignore it if not available?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/5392?email_source=notifications&email_token=AAA6QZGNL7JIHTI4UGFL7M3PZ7YZBA5CNFSM4HWYKHBKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXOFE2Q#issuecomment-500978282>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA6QZGW4LTHIAKU4QSBG33PZ7YZBANCNFSM4HWYKHBA>
> .
>
